spec/01-unit/01-db/01-schema/01-schema_spec.lua

```diff
@@ -2989,4 +3015,249 @@ describe("schema", function()
+              return { name = name:upper() }
+            end,
+          },
+        },
+      }
+      local entity = { name = "test1" }
+      local input = { name = "we have a value" }
+
+      local TestEntities = Schema.new(test_schema)
+      local transformed_entity, _ = TestEntities:transform(entity, input)
+
+      assert.truthy(transformed_entity)
+      assert.equal("TEST1", transformed_entity.name)
+    end)
+
+    it("skips transformation when none of input matches", function()
+      local test_schema = {
+        name = "test",
+        fields = {
+          {
+            name = {
+              type = "string"
+            },
+          },
+        },
+        transformations = {
+          {
+            input = { "non_existent" },
+            func = function(non_existent)
+              return { name = non_existent:upper() }
+            end,
+          },
+        },
+      }
+      local entity = { name = "test1" }
+
+      local TestEntities = Schema.new(test_schema)
+      local transformed_entity, _ = TestEntities:transform(entity)
+
+      assert.truthy(transformed_entity)
+      assert.equal("test1", transformed_entity.name)
+    end)
+
+    it("skips transformation when none of input matches using input table", function()
+      local test_schema = {
+        name = "test",
+        fields = {
+          {
+            name = {
+              type = "string"
+            },
+          },
+        },
+        transformations = {
+          {
+            input = { "name" },
+            func = function(non_existent)
+              return { name = non_existent:upper() }
+            end,
+          },
+        },
+      }
+      local entity = { name = "test1" }
+      local input = { name = nil }
+
+      local TestEntities = Schema.new(test_schema)
+      local transformed_entity, _ = TestEntities:transform(entity, input)
+
+      assert.truthy(transformed_entity)
+      assert.equal("test1", transformed_entity.name)
+    end)
+
+
+    it("transforms fields with multiple transformations", function()
+      local test_schema = {
+        name = "test",
+        fields = {
+          {
+            name = {
+              type = "string"
+            },
+          },
+        },
+        transformations = {
+          {
+            input = { "name" },
+            func = function(name)
+              return { name = "How are you " .. name }
+            end,
+          },
+          {
+            input = { "name" },
+            func = function(name)
+              return { name = name .. "?" }
+            end,
+          },
+        },
+      }
+
+      local entity = { name = "Bob" }
+
+      local TestEntities = Schema.new(test_schema)
+      local transformed_entity, _ = TestEntities:transform(entity)
+
+      assert.truthy(transformed_entity)
+      assert.equal("How are you Bob?", transformed_entity.name)
+    end)
+
+    it("returns error if transformation returns an error", function()
```

Comment:
I think one more tests similar to this one is needed, but for a transformation function which (erroneously) returns a non-table (like a number, or a string). This case should be detected and handled too.

Reply:
this is far from the only thing that a function can do, e.g. it can:
1. yield
2. error
3. call ngx.exit (or something similar)
4. return garbage
5. halt execution
6. os.exit

So, I am not sure do we need to handle that? For better error messages on Nginx logs? If we handle that should we also handle some other cases mentioned above?